### PR TITLE
1.2.0 dev

### PR DESCRIPTION
**Pull request by ckloote**

### CODE REVIEW

File: perl-lib/OESS/lib/OESS/MPLS/Discovery/Interface.pm [hunk 1]
```diff
@@ -35,8 +35,34 @@ sub new{
 sub process_results{
     my $self = shift;
     my %params = @_;
+    my $node_name = $params{'node'};
+    my $interfaces = $params{'interfaces'};
+    foreach my $interface (@$interfaces) {
+	my $interface_id = $self->{'db'}->get_interface_id_by_names(node => $node_name, interface => $interface->{'name'});
+	if (!defined($interface_id)) {
+	    $self->{'logger'}->warn($self->{'db'}->{'error'});
```
Comment (ajragusa) [inline perl-lib/OESS/lib/OESS/MPLS/Discovery/Interface.pm]: If we have detected a new interface we need to add it to the DB for future use

File: perl-lib/OESS/lib/OESS/MPLS/Discovery/Interface.pm [hunk 2]
```diff
@@ -35,8 +35,34 @@ sub new{
...
+	}
+
+	if ($intf->{'operational_state'} ne $interface->{'operational_state'}) {
+	    my $result = $self->{'db'}->update_interface_operational_state(
+		interface_id => $interface_id,
+		operational_state => $interface->{'operational_state'}
+		);
+	    if (!defined($result)) {
+		$self->{'logger'}->warn($self->{'db'}->{'error'});
+		return;
```
Comment (ajragusa) [inline perl-lib/OESS/lib/OESS/MPLS/Discovery/Interface.pm]: We should probably attempt to do all of this in a transaction

File: perl-lib/OESS/t/MPLS-Discovery-Interface.t [hunk 3]
```diff
@@ -58,6 +58,23 @@ my $res = $interface_discovery->process_results( node => $example_node, interfac
...
+	push(@results, "ok");
+    }
+}
+ok(scalar @results == 3, "DB matches test data");
+
+@$example_data[0]->{'name'} = "fooooo!";
+$res = $interface_discovery->process_results( node => $example_node, interfaces => $example_data );
+is($res, undef, "nonexistent interface fails");
+
 #TODO:
```
Comment (ajragusa) [inline perl-lib/OESS/t/MPLS-Discovery-Interface.t]: For our tests here can we add a few tests that will set the operational state to up and then verify that the operational state is now up... and also add an additional interface and verify it populates properly

File: perl-lib/OESS/lib/OESS/MPLS/Discovery/Interface.pm [hunk 4]
```diff
@@ -35,8 +35,59 @@ sub new{
 sub process_results{
     my $self = shift;
     my %params = @_;
+    my $node_name = $params{'node'};
+    my $interfaces = $params{'interfaces'};
+    foreach my $interface (@$interfaces) {
+	my $interface_id = $self->{'db'}->get_interface_id_by_names(node => $node_name, interface => $interface->{'name'});
+	if (!defined($interface_id)) {
+	    $self->{'db'}->_start_transaction();
```
Comment (ajragusa) [inline perl-lib/OESS/lib/OESS/MPLS/Discovery/Interface.pm]: So I really want to wrap the entire processing in a transaction... so We should move the _start_transaction to before the foreach statement.... and the commit to after it

Comment (ckloote) [inline perl-lib/OESS/lib/OESS/MPLS/Discovery/Interface.pm]: Gotcha, so the whole batch of interfaces are updated atomically.

**Comment by ajragusa** [comment]
Thanks CJ

**Pull request merged**